toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -398,7 +398,19 @@ func CreateSinglePartition(diskDevPath string, partitionNumber int, partitionTab
 			return "", err
 		}
 	}
-
+	// Update kernel partition table information
+	//
+	// There can be a timing issue where partition creation finishes but the
+	// devtmpfs files are not populated in time for partition initialization.
+	// So to deal with this, we call partprobe here to query and flush the
+	// partition table information, which should enforce that the devtmpfs
+	// files are created when partprobe returns control.
+	stdout, stderr, err := shell.Execute("partprobe", "-s", diskDevPath)
```

Comment:
Which package does partprobe come from? We probably should keep track of external programs that we use...

Reply:
`partprobe` comes from the `parted` package. Any recommendations on how best to track these external dependencies?

BTW I also have #731 available, which introduces a new `installutils.GetRequiredPackagesForInstall()`. Idea being that the packages returned from this list are required for running image creation. Currently this is a Mariner-specific package list. But we could maybe extend this idea to incorporate external dependencies for all our tools?

Reply:
Not really. Preferably we'd package our toolkit as an RPM package and write Requires there ;)